OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnCreate.cs

```diff
@@ -0,0 +1,48 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits.Conditions
+{
+    [Desc("Grants a random condition from a predefined list to the actor when created.")]
+    public class GrantConditionOnCreateInfo : ITraitInfo
+    {
+        [FieldLoader.Require]
+        [GrantedConditionReference]
+        [Desc("List of condition to grant from.")]
+        public readonly string[] Conditions = null;
+
+        public object Create(ActorInitializer init) { return new GrantConditionOnCreate(init.Self, this); }
+    }
+
+    public class GrantConditionOnCreate : INotifyCreated
+    {
+        readonly GrantConditionOnCreateInfo info;
+
+        public GrantConditionOnCreate(Actor self, GrantConditionOnCreateInfo info)
+        {
+            this.info = info;
+        }
+
+        public void Created(Actor self)
+        {
+            if (!info.Conditions.Any())
```

Comment:
Does this even make sense if we have `[FieldLoader.Require]` on that field?

Reply:
No, I tested that. If you specify in the YAML
`Conditions:` and leave it blank the `Required` check passes, but there is still no value.